Userland/Libraries/LibWeb/CSS/StyleComputer.cpp

```diff
@@ -1870,6 +1885,47 @@ RefPtr<Gfx::Font const> StyleComputer::compute_font_for_style_values(DOM::Elemen
         found_font = find_font(font_family.as_custom_ident().custom_ident());
     }
 
+    // 5.2.6 If no matching face exists or the matched face does not contain a glyph for the character to be
+    // rendered, the next family name is selected and the previous three steps repeated. Glyphs from other
+    // faces in the family are not considered. The only exception is that user agents may optionally substitute
+    // a synthetically obliqued version of the default face if that face supports a given glyph and synthesis
+    // of these faces is permitted by the value of the font-synthesis property. For example, a synthetic italic version
+    // of the regular face might be used if the italic face doesn’t support glyphs for Arabic.
+    auto face_contains_glyphs = [&](RefPtr<Gfx::Font const> font, FlyString const& text) -> bool {
+        if (font) {
+            for (auto code_point : text.code_points()) {
+                if (!is_ascii_control(code_point) && !font->contains_glyph(code_point))
```

Comment:
Skipping control code points makes sense, but it's not mentioned in the spec text here. Is it supposed to happen somewhere else?

Also, Unicode has a lot of other control code points we should treat the same way. I think the check would be `Unicode::code_point_has_general_category(code_point, Unicode::GeneralCategory::Control)`? Pinging @trflynn89

Reply:
Exactly - but you can't access the enum directly, because we can't assume Unicode data generation is enabled. So something like:
```c++
static auto control_general_category = Unicode::general_category_from_string("Control"sv);

if (control_general_category.has_value() && Unicode::code_point_has_general_category(code_point, *control_general_category)) {
	// ...
}
```

Reply:
Thank you for the recommendation @trflynn89. I have updated the check for control code points using the above Unicode method.

@AtkinsSJ The spec doesn't mention control characters, and I believe if this was done per glyph as mentioned by @awesomekling it would be in a place that would not have control characters. So I think that's why the spec doesn't mention it. However, as you mentioned, it would not be very easy to do it per glyph because it requires the significant changes.